tests/util/base.py

```diff
@@ -37,6 +38,22 @@ def dbloader(config):
     setup_test_db()
 
 
+@yield_fixture
+def patched_gevent_sleep(monkeypatch):
+    monkeypatch.setattr('gevent.sleep', mock.Mock())
+    yield
+    monkeypatch.undo()
+
+
+@yield_fixture
+def patched_gevent(monkeypatch):
+    monkeypatch.setattr('gevent.sleep', mock.Mock())
+    monkeypatch.setattr('gevent.spawn', mock.Mock())
```

Comment:
We do this because the test had crashing greenlets due to the fact that MockIMAPClient doesn't implement some functionality that wasn't part of the test (something related to IDLE).